Gems/ROS2/Code/Include/ROS2/Manipulator/MotorizedJointComponent.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "MotorizedJointBus.h"
+#include "ROS2/VehicleDynamics/DriveModels/PidConfiguration.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Math/Vector2.h>
+
+namespace ROS2
+{
+    //! A prototype component for simulated joint with a motor.
+    //! It works with either TransformBus or RigidBodyBus.
+    //! TransformBus mode, called `AnimationMode` changes local transform. In this mode, you cannot have a rigid body
+    //! controller enabled. With RigidBodyBus it applies forces and torque according to PID control.
+    //! @note This class is already used through ROS2FrameComponent.
+    // TODO This is a prototype. Tasks: refactor, add bus interface, rotation, ramps, cascading controllers, tests.
+    class MotorizedJointComponent
+        : public AZ::Component
+        , public AZ::TickBus::Handler
+        , public MotorizedJointRequestBus::Handler
+    {
+    public:
+        AZ_COMPONENT(MotorizedJointComponent, "{AE9207DB-5B7E-4F70-A7DD-C4EAD8DD9403}", AZ::Component);
+
+        MotorizedJointComponent() = default;
+        ~MotorizedJointComponent() = default;
+        void Activate() override;
+        void Deactivate() override;
+        static void Reflect(AZ::ReflectContext* context);
+
+        //! Set a setpoint (e.g. desired local position). The controller will follow it.
+        void SetSetpoint(float setpoint) override;
+
+        //! Get a setpoint
+        float GetSetpoint() override;
+
+        //! Get current control error. It is the difference between control value and measurement.
+        //! When the setpoint is reached this should be close to zero.
+        //! @returns control error, in meters for linear joints and in radians for angular joints.
+        float GetError() override;
+
+        //! Get current position from measurement.
+        //! @returns current position, in meters for linear joints and radians for angular joints.
+        float GetCurrentMeasurement() override;
```

Comment:
A section comment block would be nice here. You can omit documenting the overrides as well
```suggestion
        //////////////////////////////////////////////////////////////////////// 
        // MotorizedJointRequestBus::Handler overrides
        void SetSetpoint(float setpoint) override;
        float GetSetpoint() override;
        float GetError() override;
        float GetCurrentMeasurement() override;
        //////////////////////////////////////////////////////////////////////// 
```